src/DeploymentProvider/AzureSpringCloudDeploymentProvider.ts

```diff
@@ -0,0 +1,207 @@
+            }
+            this.params.ResourceGroupName = filteredResources[0].id.substring(beginIndex, endIndex);
+            core.debug("resource group name: " + this.params.ResourceGroupName);
+        }
+        else { //Should never ever ever happen
+            throw new Error('DuplicateAzureSpringCloudName');
+        }
+        const serviceResponse = await this.client.services.get(this.params.ResourceGroupName, this.params.ServiceName);
+        core.debug("service response: " + serviceResponse._response.bodyAsText);
+    }
+
+    public async DeployAppStep() {
+        switch (this.params.Action) {
+
+            case Actions.deploy: {
+                await this.performDeployAction();
+                break;
+            }
+
+            case Actions.setProduction: {
+                await this.performSetProductionAction();
+                break;
+            }
+
+            case Actions.deleteStagingDeployment: {
+                await this.performDeleteStagingDeploymentAction();
+                break;
+            }
+
+            default:
+                throw Error('UnknownOrUnsupportedAction' + this.params.Action);
+        }
+    }
+
+    private async performDeleteStagingDeploymentAction() {
+        console.log('Delete staging deployment action');
+        const deploymentName = await dh.getStagingDeploymentName(this.client, this.params);
+        this.params.DeploymentName = deploymentName;
+        console.log(`Action for service ${this.params.ServiceName} app ${this.params.AppName} deployment ${deploymentName}`);
+        if (deploymentName) {
+            await dh.deleteDeployment(this.client, this.params);
+        } else {
+            throw Error('NoStagingDeploymentFound');
+        }
+        console.log('Delete staging deployment action successful');
+        return deploymentName;
+    }
+
+    private async performSetProductionAction() {
+        console.log('Set production action');
+        let deploymentName: string;
+        if (this.params.UseStagingDeployment) {
+            core.debug('Targeting inactive deployment');
+            deploymentName = await dh.getStagingDeploymentName(this.client, this.params);
+            this.params.DeploymentName = deploymentName;
+            if (!deploymentName) { //If no inactive deployment exists, we cannot continue as instructed.
+                throw Error('NoStagingDeploymentFound');
+            }
+        }
+        else {
+            //Verify that the named deployment actually exists.
+            deploymentName = this.params.DeploymentName;
+            let existingStagingDeploymentName: string = await dh.getStagingDeploymentName(this.client, this.params);
+            if (deploymentName != existingStagingDeploymentName) {
+                throw Error('StagingDeploymentWithNameNotExist:' + deploymentName);
+            }
+        }
+        console.log(`Action for service ${this.params.ServiceName} app ${this.params.AppName} deployment ${deploymentName}`);
```

Comment:
Set production deployment for service ${this.params.ServiceName} app ${this.params.AppName} to deployment ${deploymentName}